controllers/datadogagent/merger/service.go

```diff
@@ -0,0 +1,80 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```

Comment:
I think it'd be good to add tests for the functions exposed in this file.

We can take as an example the tests that we have for the other files in this package (`envvars_test.go`, `rbac_test.go`, etc.)